### PR TITLE
Consequetive chat cell, Time label and Scroll to latest msg implemented

### PR DESCRIPTION
**Pull request by hasanmainul**

### CODE REVIEW

File: LocalChat/ChatViewController.swift [hunk 1]
```diff
@@ -42,8 +40,17 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
             userSegmentControl.insertSegmentWithTitle("BOT\(count + 1)", atIndex: count, animated: true)
         }
         userSegmentControl.selectedSegmentIndex = 0;
+        chatUserNameLabel?.text = userSegmentControl.titleForSegmentAtIndex(userSegmentControl.selectedSegmentIndex) as String!
         
         chatTableView.reloadData()
+        scrollToBottom()
+    }
+    
+
```
Comment (ahammad-reyad) [inline LocalChat/ChatViewController.swift:59]: remove extra line

File: LocalChat/ChatViewController.swift [hunk 2]
```diff
@@ -74,10 +81,14 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
-        self.chatMessages = ChatManager.sharedInstance.sendChatMessage(chatTextField.text!, withUserName: userName)
-        chatText = chatTextField.text!
-        chatTextField?.text = ""
-        self.chatTableView.reloadData()
+        
+         if chatTextField?.text?.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet()).characters.count != 0 {
+            self.chatMessages = ChatManager.sharedInstance.sendChatMessage(chatTextField.text!, withUserName: userName, sentTime: NSDate.init())
+            chatText = chatTextField.text!
+            chatTextField?.text = ""
+                self.chatTableView.reloadData()
```
Comment (ahammad-reyad) [inline LocalChat/ChatViewController.swift]: indent

File: LocalChat/ChatViewController.swift [hunk 3]
```diff
@@ -16,7 +16,7 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
     @IBOutlet weak var chatTextField: UITextField!
     
     var frameView: UIView!
-    var numberOfSegment: Int = Constants.segment.minimumNumberOfSegments
+    var numberOfSegment: Int = Constants.user.minimumNumberOfUsers
     var chatText: String!
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: If not needed remove chatText

File: LocalChat/ChatViewController.swift (hunk 2 above)
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Organize your code using MARK. Like - 
// MARK: – Segment Control
// MARK: – Handling Keyboard
// MARK: – TextField delegate or TextField datasource
// MARK: – TableView delegate or TableView datasource
Help : http://www.learnswiftonline.com/reference-guides/organizing-your-swift-code/

**Comment by jannatulabeda** [comment]
For LocalChat/ChatViewController.swift and LocalChat/SettingViewController.swift
Organize your code using MARK. Like - 
// MARK: – Segment Control
// MARK: – Handling Keyboard
// MARK: – TextField delegate or TextField datasource
// MARK: – TableView delegate or TableView datasource 
// MARK: – Button pressed event etc.
Help : http://www.learnswiftonline.com/reference-guides/organizing-your-swift-code/

File: LocalChat/LocalChat/Controllers/ChatViewController.swift [hunk 4]
```diff
@@ -66,50 +102,62 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
         super.didReceiveMemoryWarning()
```
Comment (jannatulabeda) [inline LocalChat/LocalChat/Controllers/ChatViewController.swift]: This method is inside // MARK: - Keyboard Handling
Please fix this.

**Pull request merged**